### PR TITLE
Work around https://github.com/pypa/pip/pull/9450

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
Work around https://github.com/pypa/pip/pull/9450
See https://github.com/pypa/pip/issues/10212

### CODE REVIEW

**Comment by s0undt3ch** [comment]
```python
❯ python
Python 3.6.8 (default, Apr 15 2020, 09:32:35) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import logging
>>> logging.getLoggerClass()
<class 'logging.Logger'>
>>> import salt
>>> logging.getLoggerClass()
<class 'salt._logging.impl.SaltLoggingClass'>
>>> logging.getLogger("foo").verbose("blah")
>>>
```
Works in _theory_...

**Comment by s0undt3ch** [comment]
re-run full all

**Comment by s0undt3ch** [comment]
re-run full all

File: salt/__init__.py [hunk 1]
```diff
@@ -119,3 +119,6 @@ def __define_global_system_encoding_variable__():
 
 # This is now garbage collectable
 del __define_global_system_encoding_variable__
+
+# Import Salt's logging machinery
+import salt._logging.impl  # isort:skip  pylint: disable=unused-import
```
Comment (dwoz) [inline salt/__init__.py:124]: What happens for older versions of pip? Not all OSes will have the newest version, right?

Comment (s0undt3ch) [reply salt/__init__.py:124]: Nothing, we hack around if we need to. On older versions we don't do anything.

Comment (s0undt3ch) [reply salt/__init__.py:124]: https://github.com/saltstack/salt/pull/60625/files#diff-721c42a3a37808f0a965af6afc0d1f17dc854895084f1071bbd00fe9dcfceed0R400
That module only exists on versions that need the work around.

**Review by krionbsd — APPROVED** [review]

**Review by Ch3LL — APPROVED** [review]

**Review by dwoz — APPROVED** [review]

**Review by garethgreenaway — APPROVED** [review]

**Comment by gediminasz** [comment]
I've encountered a possible related error on Ubuntu 22.04 and Python 3.10.4:

```
----------
          ID: docker_package_installed
    Function: pip.installed
        Name: docker
      Result: True
     Comment: Python package docker was already installed
              All specified packages are already installed
     Started: 11:34:53.021563
    Duration: 1240.316 ms
     Changes:   
----------
          ID: docker_network_present
    Function: docker_network.present
        Name: test
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python3/dist-packages/salt/state.py", line 2179, in call
                  ret = self.states[cdata["full"]](
                File "/usr/lib/python3/dist-packages/salt/loader/lazy.py", line 149, in __call__
                  return self.loader.run(run_func, *args, **kwargs)
                File "/usr/lib/python3/dist-packages/salt/loader/lazy.py", line 1201, in run
                  return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
                File "/usr/lib/python3/dist-packages/salt/loader/lazy.py", line 1216, in _run_as
                  return _func_or_method(*args, **kwargs)
                File "/usr/lib/python3/dist-packages/salt/loader/lazy.py", line 1249, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python3/dist-packages/salt/states/docker_network.py", line 625, in present
                  log.trace("Details of docker network '%s': %s", name, network)
              AttributeError: 'VerboseLogger' object has no attribute 'trace'
     Started: 11:34:54.263015
    Duration: 6.73 ms
     Changes:
```

States:

```yml
docker_package_installed:
  pip.installed:
    - name: docker

docker_network_present:
  docker_network.present:
    - name: test
```

Succeeds on first run but on second run (when docker network is already present) it fails with the error above.

**Pull request merged**